SplitScreenScanner/Classes/ContinuousBarcodeScanner.swift

```diff
@@ -116,8 +163,9 @@ extension ContinuousBarcodeScanner {
         captureSession.startRunning()
 
         let rectOfInterest = previewLayer.metadataOutputRectConverted(fromLayerRect: previewView.bounds)
-        metadataCapture.startRunning(rectOfInterest: rectOfInterest) { [weak self] barcode in
-            self?.delegate?.didScan(barcode: barcode)
+        metadataCapture.startRunning(rectOfInterest: rectOfInterest) { [weak self] barcodeObject in
+            self?.delegate?.didScan(barcode: barcodeObject.stringValue ?? "")
+            self?.addBarcodeOverlayViewFor(barcodeObject: barcodeObject)
```

Comment:
Do we need to dispatch to the main thread here?

Reply:
The whole closure is already dispatched to the main thread via line 51 👍